sklearn/preprocessing/mice.py

```diff
@@ -0,0 +1,365 @@
+        the current column. Can provide significant speed-up
+        when the number of columns is huge.
+
+    initial_fill_method : str, optional, default = "mean"
+        Valid values: {"mean", "median", or "most_frequent"}.
+        Uses sklearn.preprocessing.Imputer.
+
+    min_value : float, default = None
+        Minimum possible imputed value.
+
+    max_value : float, default = None
+        Maximum possible imputed value.
+
+    verbose : boolean, default = False
+        Whether to provide updates on progress.
+
+    Notes
+    -----
+    - Columns which only contain missing values at `fit` are discarded upon
+        `transform`.
+
+    References
+    ----------
+
+    .. [1] `Stef van Buuren, Karin Groothuis-Oudshoorn (2011). "mice:
+        Multivariate Imputation by Chained Equations in R". Journal of
+        Statistical Software 45: 1-67.
+        <https://www.jstatsoft.org/article/view/v045i03>`_
+    """
+
+    def __init__(
+            self,
+            missing_values='NaN',
+            imputation_order='monotone',
+            n_imputations=100,
+            n_burn_in=10,
+            n_nearest_columns=np.infty,
+            initial_fill_method="mean",
+            min_value=None,
+            max_value=None,
+            verbose=False):
+        from ..linear_model import BayesianRidge  # avoiding circular import
+        self.model = BayesianRidge()
+        self.missing_values = missing_values
+        self.imputation_order = imputation_order
+        self.n_imputations = n_imputations
+        self.n_burn_in = n_burn_in
+        self.n_nearest_columns = n_nearest_columns
+        self.initial_fill_method = initial_fill_method
+        self.min_value = min_value
+        self.max_value = max_value
+        self.verbose = verbose
+        self.trained_model_triplets = []
+
+    def _fill_in_one_column(self,
+                            X_filled,
+                            mask_missing_values,
+                            this_column,
+                            other_columns,
+                            model=None,
+                            min_std=1e-6):
+        # if nothing is missing, just return the default
+        if mask_missing_values[:, this_column].sum() == 0:
+            return X_filled, model
+        missing_row_mask = mask_missing_values[:, this_column]
+        if model is None:
+            X_train = X_filled[:, other_columns][~missing_row_mask]
+            y_train = X_filled[:, this_column][~missing_row_mask]
+            if np.std(y_train) > 0:
+                model = deepcopy(self.model)
+                model.fit(X_train, y_train)
+            else:
+                model = DummyRegressor()
+                model.fit(X_train, y_train)
+        X_test = X_filled[:, other_columns][missing_row_mask]
+        mus, sigmas = model.predict(X_test, return_std=True)
+        sigmas = np.maximum(sigmas, min_std)
+        imputed_values = np.random.normal(mus, sigmas)
```

Comment:
at the beginning of the function get the random state `rng = check_random_state(self.random_state)` and use it here `rng.random.normal(mus, sigmas)`